### ISSUE TITLE
Question about the corresponding point matching

### ISSUE THREAD

**Issue by jizongFox** [RaymondJiangkw/COGS#14]
Hello, 

Thank you for this interesting work. Based on my shallow understanding, one uses the matched points to "drag" the camera poses to their right positions. Certainly, this works and it reminds me of a paper https://arxiv.org/abs/2312.09031 using the similar idea. In this referenced paper, the authors use a pretrained corresponding model (similar to yours) and directly optimize the camera pose, without any back-projection. 

I would be much appreciated if you could provide any insight about the advantage of your proposed method, compared with this simple baseline. Any thought about which one to use in which scenario?

Thank you and look forward to your reply. 

Jizong Peng

**Comment by RaymondJiangkw** [comment]
Hi, thanks for your interest. I am not aware of this paper until you mention it here, and it seems interesting.

Specifically, if I understand correctly, our paper and your referenced paper are addressing different problems.

For your referenced paper, it addresses the problem of estimating the camera pose for a view given the reconstructed scene, which is therefore assumed to be known.

For us, we both recover the scene and estimate the camera poses simultaneously, which introduces more challenges, especially in the sparse view case. In practice, a well-reconstructed scene is not always assumed, especially when there are only sparse observations. Our paper is proposed to address this specific need.

If you have a well-represented scene, I think then you don't need to apply our algorithm to recover the scene as well, but only need to solve for the camera poses using their paper. :)

Hope you can find this helpful.

**Comment by jizongFox** [comment]
That makes sense, thank you very much for the insightful response

**Comment by jizongFox** [comment]
Hello @RaymondJiangkw 

Thank you for answering my previous question. 

I still have one question regarding the idea of your methodology and would appreciate it if you could share your insight. 

This paper uses the correspondences to "drag" camera poses with your modified rasterizer. While the intuition is quite direct, I am wondering if there's an alternative solution in this context.  For example,  given corresponding points between two images, one can compute the Fundamental (F) matrix, which is fixed and as a ground true to tell the relationship between two images. (In the paper, you compute the correpondances between rendered image and the gt. While computing F between two adjacent ground truth images can give an accurate description of the poses). In this way, the pose can also be regularized by this ground truth F, as the F can either be computed from corresponding points, or from relative poses (up to a scale). 

Does this alternative make any sense?

Jizong Peng

**Comment by RaymondJiangkw** [comment]
Hi, thanks for your interest. We indeed thought about this before, and validated this idea in the paper.

On the one hand, from the perspective of estimating the camera pose, if you have very accurate correspondences, that would be ideal. You even don't really need many correspondences to calculate the relative pose. However, the main problem is that, in the in-the-wild cases, and especially for sparse views, very accurate correspondences are usually not assumed. The detected correspondences contain many outliers that require careful treatment through method such as RANSAC. We indeed try to directly use perspective-n-points to get the camera pose, as it also gives a differentiable estimation. However, as shown in the Fig. 4, it gives wrong camera pose due to the outliers, which leads to catastrophic failure. In contrast, the gradient descent method which we propose is much more robust to the outliers.

On the other hand, we are also using the correspondences to help us correct the monocular depth, which is possible due to our differentiable pipeline to optimize over the screen-space points. Fundamental matrix cannot directly help with this. :)

If you really want to look for alternative choices, perhaps you will find pixelSplat, or instantSplat interesting, as they leverage Dust3R to estimate dense point clouds and relative pose from a pair of images, using a pretrained neural network. They also work well in the sparse view case.

**Comment by jizongFox** [comment]
Hello @RaymondJiangkw, 

Thank you a lot for the kind explanation. I am curious about how you compute the camera poses using PnP algorithm. I am a bit confused about the fact that PnP is designed for find a camera pose from N 3D point to N 2D correspondences. In this context, you might have projected gaussian point back to 3D space with the mono-depth, and using that 3D world coordinate to find the next image's pose?

I am not sure about the details of the algorithm, and I would be much appreciated if you could give any hint about this.

Best,
Jizong Peng

**Comment by RaymondJiangkw** [comment]
Hi, the specific scheme is that we try to use the PnP to estimate the camera pose of next unregistered view, relative to our current reconstructed scene.

So, the next unregistered view is initialized at the last registered view, for which we can render the depth map and rgb map. Given the detected correspondences between rendered rgb map at the last registered view and the ground-truth rgb map of next unregistered view, the correspondent points on the rendered rgb map are therefore lifted to 3D points using the rendered depth map.

And the PnP is then applied to the lifted 3D points from the last registered view, and the 2D correspondent points from the next unregistered view.
